Framework/Subset_of_CorLib/System/Number.cs

```diff
@@ -371,7 +371,7 @@ private static String PostProcessInteger(Object value, String original, char for
                     {
                         int len = result.Length;
 
-                        if(value is sbyte)
+                        if(value is sbyte || value is byte)

```

Comment:
I think this is because this "post processing" is actually fixing wrong native output in the case of byte. Anyone reading this fixed code would question the ushort case too.

Might be worth fixing the original cause, e.g. in [corlib_native_System_Number.cpp](../blob/dev/CLR/Libraries/CorLib/corlib_native_System_Number.cpp) changing `char` to `unsigned char` unless there is a compiler reason not to (works for me in the emulator).
```cpp
case DATATYPE_U1:  
hal_snprintf( result, ARRAYSIZE(result), "%.*X", precision,
                                               (unsigned char)value->NumericByRef().u1 );  
break; 
```